pkg/cmd/roachtest/test_runner.go

```diff
@@ -605,17 +605,22 @@ func (r *testRunner) runWorker(
 			qp.Release(alloc)
 		}
 	}()
-
+	// cluster destroy can be done concurrently. We just need to make sure that we exit the roachtest once all
+	// the cluster destroy operations are complete!
+	clusterDestroyWg := &sync.WaitGroup{}
+	defer clusterDestroyWg.Wait() // wait for the clusters to be destroyed
 	// Loop until there's no more work in the pool, we get interrupted, or an
 	// error occurs.
 	for {
 		select {
 		case <-interrupt:
 			l.ErrorfCtx(ctx, "worker detected interruption")
+			r.destroyCluster(ctx, clusterDestroyWg, c, l, true)
```

Comment:
nit: if you do keep the `wait` flag, then add a `/* Wait */` comment.

Reply:
wait is removed as per your suggestion